File: frontend/src/types/ResourceStatus.ts
export type ResourceStatus = "AVAILABLE" | "BOOKED" | "MAINTENANCE" | "OFFLINE";

export const RESOURCE_STATUSES: ResourceStatus[] = [
  "AVAILABLE",
  "BOOKED",
  "MAINTENANCE",
  "OFFLINE",
];

export const RESOURCE_STATUS_TEXT: Record<ResourceStatus, string> = {
  AVAILABLE: "可用",
  BOOKED: "占用中",
  MAINTENANCE: "维护中",
  OFFLINE: "离线",
};

export type BookingStatus = "CONFIRMED" | "PENDING" | "RELEASED";

export const BOOKING_STATUS_TEXT: Record<BookingStatus, string> = {
  CONFIRMED: "已确认",
  PENDING: "待处理",
  RELEASED: "已释放",
};

export type GroundTaskStatus = "PLANNED" | "ACCEPTED" | "COMPLETED" | "BLOCKED";

export const GROUND_TASK_STATUS_TEXT: Record<GroundTaskStatus, string> = {
  PLANNED: "待签收",
  ACCEPTED: "已签收",
  COMPLETED: "已完成",
  BLOCKED: "阻塞",
};

File: backend/src/constants/ResourceStatus.go
package constants

// ResourceStatus values, duplicated in frontend constants/ResourceStatus.
const (
	ResourceAvailable   = "AVAILABLE"
	ResourceBooked      = "BOOKED"
	ResourceMaintenance = "MAINTENANCE"
	ResourceOffline     = "OFFLINE"
)

var ResourceStatus = []string{
	ResourceAvailable,
	ResourceBooked,
	ResourceMaintenance,
	ResourceOffline,
}

var ResourceStatusText = map[string]string{
	ResourceAvailable:   "可用",
	ResourceBooked:      "占用中",
	ResourceMaintenance: "维护中",
	ResourceOffline:     "离线",
}

func IsValidResourceStatus(value string) bool {
	for _, item := range ResourceStatus {
		if item == value {
			return true
		}
	}
	return false
}

// Resource booking status values (ResourceBooking.booking_status).
const (
	BookingConfirmed = "CONFIRMED"
	BookingPending   = "PENDING"
	BookingReleased  = "RELEASED"
)

var BookingStatus = []string{BookingConfirmed, BookingPending, BookingReleased}

var BookingStatusText = map[string]string{
	BookingConfirmed: "已确认",
	BookingPending:   "待处理",
	BookingReleased:  "已释放",
}

// Ground task lifecycle status values.
const (
	TaskPlanned   = "PLANNED"
	TaskAccepted  = "ACCEPTED"
	TaskCompleted = "COMPLETED"
	TaskBlocked   = "BLOCKED"
)

var GroundTaskStatus = []string{TaskPlanned, TaskAccepted, TaskCompleted, TaskBlocked}

var GroundTaskStatusText = map[string]string{
	TaskPlanned:   "待签收",
	TaskAccepted:  "已签收",
	TaskCompleted: "已完成",
	TaskBlocked:   "阻塞",
}
